packages/relay-compiler/handlers/viewer/__tests__/RelayViewerHandleTransform-test.js

```diff
@@ -48,4 +53,31 @@ describe('RelayViewerHandleTransform', () => {
       return documents.join('\n');
     });
   });
+
+  it('ignores schemas where viewer has an id', () => {
+    const schema = buildASTSchema(parse(`
+      type Query {
+        viewer: Viewer
+      }
+      type Viewer {
+        id: ID!
+      }
```

Comment:
For example, the graph.cool service generates schemas where `Viewer` is a User object that has an `id`.